Dockerfile

```diff
@@ -1,22 +1,67 @@
-FROM node:22-bookworm-slim
+ARG USERNAME=node
+ARG PLAYWRIGHT_BROWSERS_PATH=/ms-playwright
+
+# ------------------------------
+# Base
+# ------------------------------
+FROM node:22-bookworm-slim AS base
 
 # Set the working directory
 WORKDIR /app
 
-# Copy package.json and package-lock.json at this stage to leverage the build cache
-COPY package*.json ./
+RUN --mount=type=cache,target=/root/.npm,sharing=locked,id=npm-cache \
+    --mount=type=bind,source=package.json,target=package.json \
+    --mount=type=bind,source=package-lock.json,target=package-lock.json \
+  # Install `node_modules` including dev dependencies to run playwright commands with npx
+  npm ci && \
+  # Install system dependencies for playwright
+  npx playwright install-deps chromium-headless-shell && \
+  # Overwrite with only production `node_modules` required for runtime
+  npm ci --omit=dev
+
+# ------------------------------
+# Builder
+# ------------------------------
+FROM base AS builder
 
-# Install dependencies
-RUN npm ci
+ARG PLAYWRIGHT_BROWSERS_PATH
+ENV PLAYWRIGHT_BROWSERS_PATH=${PLAYWRIGHT_BROWSERS_PATH}
 
-# Install chromium and its dependencies, but only for headless mode
-RUN npx -y playwright install --with-deps --only-shell chromium
+RUN --mount=type=cache,target=/root/.npm,sharing=locked,id=npm-cache \
+    --mount=type=bind,source=package.json,target=package.json \
+    --mount=type=bind,source=package-lock.json,target=package-lock.json \
+  # Install `node_modules` including dev dependencies to run playwright commands with npx
+  npm ci && \
+  # Install browser binaries
+  npx playwright install --only-shell chromium
```

Comment:
`--only-shell` is incompatible with the server as we are using new headless (from chromium) on ToT.

Reply:
@pavelfeldman 
Thank you for your reply!

I’ve updated the browser executable being installed to use the new headless mode.
Since the new headless mode requires specifying the `chromium` channel in `LaunchOptions`, I also made a small change to the source code.